src/VisualStudio/IntegrationTest/New.IntegrationTests/CSharp/CSharpFindReferences.cs

```diff
@@ -76,8 +76,8 @@ void M()
             await WaitForNavigateAsync(HangMitigatingCancellationToken);
 
             // Assert we are in the right file now
-            var dirtyModifier = await TestServices.Editor.GetDirtyIndicatorAsync(HangMitigatingCancellationToken);
-            Assert.Equal($"Class1.cs{dirtyModifier}", await TestServices.Shell.GetActiveWindowCaptionAsync(HangMitigatingCancellationToken));
+            await TestServices.SolutionExplorer.SaveAllAsync(HangMitigatingCancellationToken);
+            Assert.Equal($"Class1.cs", await TestServices.Shell.GetActiveWindowCaptionAsync(HangMitigatingCancellationToken));
```

Comment:
is there really no helper or easy way to just ask "what file are we in?".  it feels very wonky that these tests have to interrogate a tab to learn this info.  @jasonmalinowski is there no easy DTE way to just ask what hte active file is?

Reply:
I'm not sure why this was written this way; indeed there are several other ways to do this. DTE, ask the window frame for it's docdata...

Reply:
So what's the solution here? Should I try to find some other helpers in DTE to directly verify the active window? Or we proceed with this workaround?

Reply:
@Cosifne This is what I was thinking about replacing `GetDirtyIndicatorAsync` with.

```csharp
        public async Task<string> GetActiveDocumentFileNameAsync(CancellationToken cancellationToken)
        {
            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);

            var monitorSelection = await GetRequiredGlobalServiceAsync<SVsShellMonitorSelection, IVsMonitorSelection>(cancellationToken);
            ErrorHandler.ThrowOnFailure(monitorSelection.GetCurrentElementValue((uint)VSConstants.VSSELELEMID.SEID_WindowFrame, out var windowFrameObj));
            var windowFrame = (IVsWindowFrame)windowFrameObj;

            ErrorHandler.ThrowOnFailure(windowFrame.GetProperty((int)__VSFPROPID.VSFPROPID_DocData, out var docDataObj));
            var document = (ITextDocument)docDataObj.GetPropertyValue("TextDocument");
            return Path.GetFileName(document.FilePath);
        }
```

Reply:
I ran into issues with this when doing integration tests for GTD. I think getting caption is silly, and either VSFPROPID_pszMkDocument or VSFPROPID_OwnerCaption would be better.

The former should be file name, and the latter should be the portion of the caption without dirty indicators and other things the editor adds.

Reply:
I think I will try whatever way I can get the file path.
Btw, how can you guys know such black magic? Any magic book so that I can become a wizard?

Reply:
> how can you guys know such black magic?

Luck, and time server, based on what code each person has had the pleasure of encountering. I could easily ask you this same question for every single PR you've ever opened :)

For me, it's because I used to do work in the bit of CPS where these GetProperty calls end up.